flytekit/core/array_node.py

```diff
@@ -0,0 +1,214 @@
+import math
+from typing import Any, List, Optional, Set, Tuple, Union
+
+from flytekit.core import interface as flyte_interface
+from flytekit.core.context_manager import ExecutionState, FlyteContext
+from flytekit.core.interface import transform_interface_to_list_interface, transform_interface_to_typed_interface
+from flytekit.core.launch_plan import LaunchPlan
+from flytekit.core.node import Node
+from flytekit.core.promise import (
+    Promise,
+    VoidPromise,
+    flyte_entity_call_handler,
+    translate_inputs_to_literals,
+)
+from flytekit.core.task import TaskMetadata
+from flytekit.loggers import logger
+from flytekit.models import literals as _literal_models
+from flytekit.models.core import workflow as _workflow_model
+from flytekit.models.literals import Literal, LiteralCollection, Scalar
+
+
+class ArrayNode:
+    def __init__(
+        self,
+        target: LaunchPlan,
+        concurrency: Optional[int] = None,
+        min_successes: Optional[int] = None,
+        min_success_ratio: Optional[float] = None,
+        bound_inputs: Optional[Set[str]] = None,
```

Comment:
all the code related to `bound_inputs` should be removed since we can't partially eval a launchplan.

Reply:
PythonFunctionTasks will require them and we're planning on adding that as a follow up. Since this is a little more ingrained in the logic, I lean towards keeping this in as well. 

Also to note, there is a follow up task to handle bound_input support for mapping over launch plans. I don't know how soon that will become a priority add though.

Reply:
How is this going to work though? For reference, the support for partial inputs in the case of tasks works today by having the bound inputs as part of the task template.

Reply:
I'll have to look into this further. I initially was looking to update ArrayNode idl to have bounded inputs list and update the validating interface functions in the admin side compiler. I'm unsure if that's the best route. Was looking to shift this mapping to the ArrayNode level and look into keeping the underlying entity's (LaunchPlan, Task, etc) interface unchanged.

Reply:
We'll need to modify our approach to bound inputs. Right now we only store bound inputs in the task template, we'll have to shift that to a "higher-level" in IDL.